src/actions/tests/pending-breakpoints.js

```diff
@@ -0,0 +1,261 @@
+  it("when the user adds a breakpoint, a corresponding pending breakpoint should be added", async () => {
+    const { dispatch, getState } = createStore(simpleMockThreadClient);
+    const bp = generateBreakpoint("foo");
+    const id = makePendingLocationId(bp.location);
+
+    await dispatch(actions.addBreakpoint(bp.location));
+    const pendingBps = selectors.getPendingBreakpoints(getState());
+    expect(pendingBps.size).to.be(2);
+    expect(pendingBps.get(id)).to.eql(generatePendingBreakpoint(bp));
+  });
+
+  it("when the user adds a sliding breakpoint, a corresponding pending breakpoint should be added", async () => {
+    const { dispatch, getState } = createStore(slidingMockThreadClient);
+    const bp = generateBreakpoint("foo");
+    await dispatch(actions.addBreakpoint(bp.location));
+    const pendingBps = selectors.getPendingBreakpoints(getState());
+
+    const slidBp = slideMockBp(bp);
+    const newId = makePendingLocationId(slidBp.location);
+
+    const bps = selectors.getPendingBreakpoints(getState());
+    const newBp = pendingBps.get(newId);
+    expect(newBp).to.eql(generatePendingBreakpoint(slidBp));
+  });
+
+  describe("when two or more breakpoints are added", () => {
+    let breakpoint1;
+    let breakpoint2;
+    let breakpointId1;
+    let breakpointId2;
+
+    beforeEach(() => {
+      breakpoint1 = generateBreakpoint("foo");
+      breakpoint2 = generateBreakpoint("foo2");
+      breakpointId1 = makePendingLocationId(breakpoint1.location);
+      breakpointId2 = makePendingLocationId(breakpoint2.location);
+    });
+
+    it("adds a corresponding pendingBreakpoint for each new addition", async () => {
+      const { dispatch, getState } = createStore(simpleMockThreadClient);
+      await dispatch(actions.addBreakpoint(breakpoint1.location));
+      await dispatch(actions.addBreakpoint(breakpoint2.location));
+
+      const pendingBps = selectors.getPendingBreakpoints(getState());
+      expect(pendingBps.get(breakpointId1)).to.eql(
+        generatePendingBreakpoint(breakpoint1)
+      );
+      expect(pendingBps.get(breakpointId2)).to.eql(
+        generatePendingBreakpoint(breakpoint2)
+      );
+    });
+
+    it("removes a corresponding pending breakpoint for each deletion", async () => {
+      const { dispatch, getState } = createStore(simpleMockThreadClient);
+      await dispatch(actions.addBreakpoint(breakpoint1.location));
+      await dispatch(actions.addBreakpoint(breakpoint2.location));
+      await dispatch(actions.removeBreakpoint(breakpoint1.location));
+
+      const pendingBps = selectors.getPendingBreakpoints(getState());
+      expect(pendingBps.has(breakpointId1)).not.to.be(true);
+      expect(pendingBps.has(breakpointId2)).to.be(true);
+    });
+  });
+
+  it("when the user disables a breakpoint, the corresponding pending breakpoint is also disabled", async () => {
+    const { dispatch, getState } = createStore(simpleMockThreadClient);
+    const bp = generateBreakpoint("foo");
```

Comment:
these two lines(171 and 172) can be moved into a beforeEach. describe block can be `synced actions between breakpoints and pendingBreakpoints`